sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/TrampolineUtil.scala

```diff
@@ -75,4 +76,13 @@ object TrampolineUtil {
   def cleanupAnyExistingSession(): Unit = SparkSession.cleanupAnyExistingSession()
 
   def asNullable(dt: DataType): DataType = dt.asNullable
+
+  /** Return true if the Attribute passed is one of aggregates in the list */
+  def validateAggregate(agg: AttributeSet): Boolean = {
+    val distinctAggs = List("avg", "sum", "count")
+    val aggName = agg.filter(x => x.toString().substring(0, 3) != "gid")
+    distinctAggs.exists(x => {
+      aggName.head.toString().startsWith(x)
+    })
+  }
```

Comment:
May I suggest this change which is cleaner and more scala-fied
```
  /** Return true if the Attribute passed is one of aggregates in the list */
  def validateAggregate(agg: AttributeSet): Boolean = {
    val distinctAggs = List("avg", "sum", "count")
    val res = agg.toSeq.exists(
      agg => distinctAggs.exists(distinctAgg => agg.name.contains(distinctAgg)))
    println("res =" + res)
    res
  }
```
I tried it with your scala test from offline discussion and it is returning the expected result.

```
 private val partialOnlyConf = replaceHashAggMode("partial").set(
    RapidsConf.ENABLE_FLOAT_AGG.key, "true").set(RapidsConf.HAS_NANS.key, "false")
    .set(RapidsConf.EXPLAIN.key, "ALL")
  IGNORE_ORDER_ALLOW_NON_GPU_testSparkResultsAreEqual(
      "Agg expression with filter with nulls", WithoutNulls,
    execsAllowedNonGpu = Seq("HashAggregateExec", "AggregateExpression",
      "AttributeReference", "Alias", "Sum", "Count", "Max", "Min", "Average", "Cast",
      "KnownFloatingPointNormalized", "NormalizeNaNAndZero", "GreaterThan", "Literal", "If",
      "EqualTo", "First", "SortAggregateExec", "Coalesce","ExpandExec", "ProjectExec"),
    conf = partialOnlyConf) {
    frame => frame.createOrReplaceTempView("testTable")
      val res = frame.sparkSession.sql(
        s"""
           | SELECT
           |   count(distinct ints),
           |   count(distinct doubles),
           |   count(longs)
           | FROM testTable
           |   group by longs
           |""".stripMargin)
      frame.printSchema()
      res.explain()
      //res.show
      res
  }
```

The debug println shows the expected boolean returned.

Reply:
Thanks! Yes, it returns the correct results for all the cases.